library/init/meta/lean_parser.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+-/
+prelude
+import init.category.monad init.category.alternative init.meta.pexpr
+
+meta constant lean_parser : Type → Type
+
```

Comment:
What about `eq`, `and`, `or`, `prod`, `unit`, `nat`, ...? I think they should stay in the top level. Moreover, it would be an insane amount of work to patch the C++ code.